CMakeLists.txt

```diff
@@ -552,6 +561,10 @@ target_compile_options(
          > # release
          > # cuda
 )
+if (SINGULARITY_STRICT_WARNINGS)
+  target_compile_options(singularity-eos_Interface INTERFACE
+    -Wall -Werror)
```

Comment:
Are there any other warning flags we want at this point like`-Wextra` or `-Wpedantic` maybe? As I'm sure you know, `-Wall` is far from all the warnings you might want.

Reply:
IMO `-Wextra` and `-Wpedantic` are too pedantic for us. `-Wextra` will catch a bunch of macros, where we've added `;` inside the macro for our own safety/sanity. `-Wpedantic` will catch variables in function signatures that are unused, just due to consistent API. 

That said, I'm open to having my mind changed.

Reply:
Fair enough on those counts. I figure we can also just add more as we need them or want them.

Reply:
Sounds good.